pkg/clients/clients.go

```diff
@@ -0,0 +1,390 @@
+			Group:   "extensions",
+			Version: "v1beta1",
+			Kind:    string(PodSecurityPolicy),
+		}
+	case PodTemplate:
+		gvk = schema.GroupVersionKind{
+			Group:   "",
+			Version: "v1",
+			Kind:    string(PodTemplate),
+		}
+	case PriorityClass:
+		gvk = schema.GroupVersionKind{
+			Group:   "scheduling.k8s.io",
+			Version: "v1beta1",
+			Kind:    string(PriorityClass),
+		}
+	case ReplicaSet:
+		gvk = schema.GroupVersionKind{
+			Group:   "apps",
+			Version: "v1",
+			Kind:    string(ReplicaSet),
+		}
+	case ReplicationController:
+		gvk = schema.GroupVersionKind{
+			Group:   "",
+			Version: "v1",
+			Kind:    string(ReplicationController),
+		}
+	case ResourceQuota:
+		gvk = schema.GroupVersionKind{
+			Group:   "",
+			Version: "v1",
+			Kind:    string(ResourceQuota),
+		}
+	case Role:
+		gvk = schema.GroupVersionKind{
+			Group:   "rbac.authorization.k8s.io",
+			Version: "v1",
+			Kind:    string(Role),
+		}
+	case RoleBinding:
+		gvk = schema.GroupVersionKind{
+			Group:   "rbac.authorization.k8s.io",
+			Version: "v1",
+			Kind:    string(RoleBinding),
+		}
+	case Secret:
+		gvk = schema.GroupVersionKind{
+			Group:   "",
+			Version: "v1",
+			Kind:    string(Secret),
+		}
+	case Service:
+		gvk = schema.GroupVersionKind{
+			Group:   "",
+			Version: "v1",
+			Kind:    string(Service),
+		}
+	case ServiceAccount:
+		gvk = schema.GroupVersionKind{
+			Group:   "",
+			Version: "v1",
+			Kind:    string(ServiceAccount),
+		}
+	case StatefulSet:
+		gvk = schema.GroupVersionKind{
+			Group:   "apps",
+			Version: "v1",
+			Kind:    string(StatefulSet),
+		}
+	case StorageClass:
+		gvk = schema.GroupVersionKind{
+			Group:   "storage.k8s.io",
+			Version: "v1",
+			Kind:    string(StorageClass),
+		}
+	case ValidatingWebhookConfiguration:
+		gvk = schema.GroupVersionKind{
+			Group:   "admissionregistration.k8s.io",
+			Version: "v1beta1",
+			Kind:    string(ValidatingWebhookConfiguration),
+		}
+	default:
+		return nil, fmt.Errorf("invalid kind for client: %s", kind)
+	}
+
+	c, err := client.ResourceClient(gvk, namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get client: %v", err)
+	}
+
+	return c, nil
+}
+
+type DynamicClientSet struct {
+	GenericClient         dynamic.Interface
+	DiscoveryClientCached discovery.CachedDiscoveryInterface
+	RESTMapper            *restmapper.DeferredDiscoveryRESTMapper
+}
+
+func NewDynamicClientSet(clientConfig *rest.Config) (*DynamicClientSet, error) {
+	disco, err := discovery.NewDiscoveryClientForConfig(clientConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize discovery client: %v", err)
+	}
+
+	// Cache the discovery information (OpenAPI schema, etc.) so we don't have to retrieve it for
+	// every request.
+	discoCacheClient := cacheddiscovery.NewMemCacheClient(disco)
+	discoCacheClient.Invalidate()
+
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(discoCacheClient)
+
+	// Create dynamic resource client
+	client, err := dynamic.NewForConfig(clientConfig)
```

Comment:
Is this client still using the REST pool?

Reply:
It's implicit now in the dynamic client.